pie/acl/__init__.py

```diff
@@ -97,60 +97,85 @@ async def repeat(self, ctx, *, input: str):
             await ctx.reply(utils.text.sanitise(input, escape=False))
     """
 
-    def predicate(ctx: commands.Context) -> bool:
-        return acl2_function(ctx, level)
+    def predicate(action: Union[commands.Context, discord.Interaction]) -> bool:
+        if type(action) is commands.Context:
+            ctx: commands.Context = action
+            return acl2_function(
+                level=level,
+                bot=ctx.bot,
+                invoker=ctx.author,
+                command=ctx.command.qualified_name,
+                guild=ctx.guild,
+                channel=ctx.channel,
+            )
+
+        bot: Union[commands.Bot, commands.AutoShardedBot] = action.client
+        invoker: Union[discord.User, discord.Member] = action.user
+        guild: discord.Guild = action.guild
+        channel: discord.abc.Messageable = action.channel
+        command: str = action.command.qualified_name
+
+        return acl2_function(
+            level=level,
+            bot=bot,
+            invoker=invoker,
+            command=command,
+            guild=guild,
+            channel=channel,
+        )
 
     return commands.check(predicate)
 
 
 # TODO Make cachable as well?
 def acl2_function(
-    ctx: commands.Context, level: ACLevel, *, for_command: Optional[str] = None
+    level: ACLevel,
+    bot: Union[commands.Bot, commands.AutoShardedBot],
+    invoker: Union[discord.User, discord.Member],
+    command: str,
+    guild: discord.Guild = None,
+    channel: discord.abc.Messageable = None,
 ) -> bool:
     """Check function based on Access Control.
 
     Set `for_command` to perform the check for other command
     then the one being invoked.
```

Comment:
You have changed the function signature, but forgot to change the related docstring.

Reply:
Updated.